### PR TITLE
[editorial] Improve exposition of the uniformity analysis

### PR DESCRIPTION
**Pull request by alan-baker**
* Motivate the analysis at a high level in the intro section
* Promote the note about reconvergence to descriptive spec text
* Add a note about function reconvergence
* Add a note to avoid grouping uniform and non-uniform inputs in a
  composite
* Uniform naming conventions for sections

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -9076,9 +9097,16 @@ from their respective desugaring translations to [[#loop-statement|loop]] statem
...
 In [[#switch-statement|switch]], a [=default-alone clause=] block is treated exactly like a [=case clause=] with regards to uniformity.
 
-Note: If the set of behaviors (see [[#behaviors]]) for an if, switch, or loop statement is {Next}, this means that we either did not diverge within the statement, or we reconverged, so we pick the node corresponding to control flow at the start of the statement as the node corresponding to control flow at the exit of the statement.
+GPUs are incentivized to minimize the amount of [=uniform control
+flow|non-uniform control flow=]; however, the points at which invocations can
+be said to be uniform varies depending on a number of factors.
+WGSL's static analysis conservatively assumes a return to uniform control flow
+occuring at the end of if, switch and loop statements if the
+[[#behaviors|behavior]] for the statement is {Next}.
+This is modelled in the preceding table as the resulting control flow node
```
Comment (mehmetoguzderin) [inline wgsl/index.bs:9106]: Minor comment: should `modeled` have the preference? The rest of the spec tends towards such spelling, including `behavior` in this PR.

**Review by mehmetoguzderin — CHANGES_REQUESTED** [review]

File: wgsl/index.bs [hunk 2]
```diff
@@ -9076,9 +9097,16 @@ from their respective desugaring translations to [[#loop-statement|loop]] statem
 
 In [[#switch-statement|switch]], a [=default-alone clause=] block is treated exactly like a [=case clause=] with regards to uniformity.
 
-Note: If the set of behaviors (see [[#behaviors]]) for an if, switch, or loop statement is {Next}, this means that we either did not diverge within the statement, or we reconverged, so we pick the node corresponding to control flow at the start of the statement as the node corresponding to control flow at the exit of the statement.
+GPUs are incentivized to minimize the amount of [=uniform control
```
Comment (dneto0) [inline wgsl/index.bs:9100]: GPUs don't have incentives.  :-)

Suggest:

> To maximize performance, implementations often try to minimize the amount of non-uniform control flow.

File: wgsl/index.bs [hunk 3]
```diff
@@ -8600,7 +8600,27 @@ See [[#statements]] and [[#function-calls]].
...
+[[#collective-operations|Collective operations]] (e.g. barriers and
+derivatives) require coordination among different invocations running
+concurrently on the GPU.  To ensure correct and portable behavior, WGSL
+requires that these operations can be statically analyzed to not have any
+control dependencies such that a non-empty strict subset of invocations will
+execute the operation (i.e. the operation must be executed in [=uniform control
+flow=]).
+Non-uniform control dependencies arise from [[#control-flow|control flow]]
+statements whose behavior depends on [=uniform value|non-uniform values=].
+These non-uniform values can be traced back to certain sources that cannot be
```
Comment (dneto0) [inline wgsl/index.bs:8612]: nit: suggest "are not" instead of "cannot be".

File: wgsl/index.bs [hunk 4]
```diff
@@ -9076,9 +9097,16 @@ from their respective desugaring translations to [[#loop-statement|loop]] statem
 
 In [[#switch-statement|switch]], a [=default-alone clause=] block is treated exactly like a [=case clause=] with regards to uniformity.
 
-Note: If the set of behaviors (see [[#behaviors]]) for an if, switch, or loop statement is {Next}, this means that we either did not diverge within the statement, or we reconverged, so we pick the node corresponding to control flow at the start of the statement as the node corresponding to control flow at the exit of the statement.
+GPUs are incentivized to minimize the amount of [=uniform control
+flow|non-uniform control flow=]; however, the points at which invocations can
```
Comment (dneto0) [inline wgsl/index.bs:9101]: nit: split the sentence.  Semicolon --> period.  "However, ..."

File: wgsl/index.bs [hunk 5]
```diff
@@ -9076,9 +9097,16 @@ from their respective desugaring translations to [[#loop-statement|loop]] statem
...
+be said to be uniform varies depending on a number of factors.
+WGSL's static analysis conservatively assumes a return to uniform control flow
+occuring at the end of if, switch and loop statements if the
```
Comment (dneto0) [inline wgsl/index.bs:9104]: typographi:
> `if`, `switch`, and `loop`

and maybe also link them.

File: wgsl/index.bs (hunk 1 above)
Comment (dneto0) [reply wgsl/index.bs:9106]: Yeah, "modeling" and "traveling" are similar cases.  I tend to use single "l".

**Review by dneto0 — CHANGES_REQUESTED** [review]
This is a big improvement.
I have some minor wording feedback

File: wgsl/index.bs (hunk 1 above)
Comment (alan-baker) [reply wgsl/index.bs]: I unified all behavior while I was at it: behaviour -> behavior.

Comment (mehmetoguzderin) [reply wgsl/index.bs]: Thank you!

**Review by mehmetoguzderin — APPROVED** [review]

**Review by dneto0 — APPROVED** [review]
Thanks!

**Pull request merged**